doc/sonic-swss-common/ZMQ producer-consumer state table design.md

```diff
@@ -0,0 +1,60 @@
+# ZMQ producer/consumer state table design
+
+# Table of Contents
+- [Table of Contents](#table-of-contents)
+- [1 Functional Requirements](#1-functional-requirement)
+  * [1.1 Supported operations](#11-supported-operations)
+  * [1.2 ZMQ producer state table support async operation](#12-zmq-producer-state-table-support-async-operation)
+  * [1.3 ZMQ consumer state table support async operation](#13-zmq-consumer-state-table-support-async-operation)(#2-configuration-and-management-requirements)
+- [2 Design](#1-design)
+  * [2.1 diagram](#21-diagram)
+  * [2.1 flow chart](#21-flow-chart)
+- [3 References](#references)
+  * [ZMQ](#zmq)
+
+# 1 Functional Requirement
+## 1.1 Supported operations
+ - Should support following operations.
+   - Set
+   - Delete
+   - Batch set
```

Comment:
>Batch set [](http://example.com/codeflow?start=5&length=9)

Could you explain the 'Batch set/delete' operations? I see other Producer/Consumer table classes only support pipeline mode.

Reply:
Fixed, design doc updated.

Reply:
Seem batch set and batch del are new functions compared with ProducerStateTable class. Do you think the concept of pipeline could implement this same feature, so we do not add new functions?

Reply:
These are not new method ProducerStateTable  class already support them:

https://github.com/sonic-net/sonic-swss-common/blob/127b1c8e9d2c81112a1c1c3f2cc44ba668dc6d58/common/producerstatetable.h#L39


    // Batched version of set() and del().
    virtual void set(const std::vector<KeyOpFieldsValuesTuple>& values);

    virtual void del(const std::vector<std::string>& keys);

Reply:
Thanks! I realized ProducerStateTable  class already support them.